certifi/core.py

```diff
@@ -18,10 +20,24 @@ class DeprecatedBundleWarning(DeprecationWarning):
     """
 
 
-def where():
-    f = os.path.dirname(__file__)
+_cert_tempfile = None
+_cert_path = os.path.join(os.path.dirname(__file__), 'cacert.pem')
+try:
+    if not os.path.exists(_cert_path):
+        # assume we are in a zip: attempt to extract to a temp file
+        cert_data = pkgutil.get_data(__package__, 'cacert.pem')
+        if cert_data is not None:
+            _cert_tempfile = tempfile.NamedTemporaryFile(suffix='.pem')
```

Comment:
`NamedTemporaryFile` `delete` defaults to `True`, and this algorithm relies on an open file handle for the duration of the runtime invocation, and breaks if the `certifi` module is reloaded, as other modules will have the old `where()` result.

Reply:
What would be a fix for this? Making `_cert_path` a subclass of `str` and storing a reference to `_cert_tempfile` there?

Maybe this edge case doesn't need to be solved together with this issue. If the module works for you now, it will continue to work. The only case that breaks is if you're in an evironment where the packages are installed outside of standard filesystem AND you reload `certifi`.

What is the use case anyway? If the app runs from a zip file or from a compiled EXE, I don't see a need for reloading `certifi` - the module will likely not change during runtime.